homeassistant/components/binary_sensor/__init__.py

```diff
@@ -166,10 +166,11 @@ def device_class(self):
 class BinarySensorDevice(BinarySensorEntity):
     """Represent a binary sensor (for backwards compatibility)."""
 
-    def __init__(self, *args, **kwargs):
+    def __new__(cls, *args, **kwargs):
```

Comment:
I don't think we should override `__new__`. It was good as it was before.

Reply:
The linter didn't warn about no call to `__init__`, I think it's because Entity extends `abc.ABC`.
Hence moving the warning to `__new__`  instead.

Reply:
It did for me when I tried this.